opentelemetry/src/metrics/instruments/counter.rs

```diff
@@ -10,6 +10,15 @@ use std::{any::Any, convert::TryFrom};
 pub trait SyncCounter<T> {
     /// Records an increment to the counter.
     fn add(&self, value: T, attributes: &[KeyValue]);
+
+    /// Creates a child counter that's bound to a specific attribute set
+    fn bind(&self, attributes: &[KeyValue]) -> Arc<dyn BoundSyncCounter<T> + Send + Sync>;
+}
```

Comment:
I saw that in the spec, but figured we didn't need it in Rust because `UnBind()` is no different than drop.

Reply:
True but what's our drop implementation if we assume static?

Reply:
How does this affect the Bounded Measure Generator being 'static ?
If at all.

Reply:
`'static` is a lifetime, it's not the `static` keyword.  A lifetime only defines the **scope** of how long a value can last for.  In other words it defines how long a value **can** last, not how long it **will** last.  Almost all owned values have a `'static` lifetime because they can be passed to other threads and can be kept alive for the rest of the program's lifetime if you want.  However, when they go out of scope they get `drop()` called, just like any other owned value.

```rust
fn do_stuff(counter: Counter<u64>) {
    // pre_do_stuff
    { 
       let bound_counter = counter.Bind(some_attributes);
       bound_counter.add(5);
    } // <-------------------------------------------------------- Drop is here
    // other stuff
}
```

Since the bound_counter variable goes out of scope on the `Drop is here` line, the compiler calls `drop()` for it.  Since `drop()` doesn't have an explicit implementation, it just calls `drop()` on all of it's fields, which will drop the `Arc`, which will either modify the reference count, or drop the underlying bound measure if it's the last instance of it.  

I'm not really sure what you would test for.  Not only is this all built-in and is fundamental to the whole Rust lifetime system, there's no observable behavior in this PR (there *is* in the performance oriented PR I have ready to go, but not in this PR).

The only reason `'static` is defined in this code is because the `Arc` is holding a trait, not a concrete object.  That means someone could have implemented the `BoundSyncCounter<T>` trait on a shared reference, in which case it's not valid to pass around because the compiler would not be able to guarantee the shared reference doesn't outlive it's owner.  Thus `'static` is required to tell the compiler "this is safe to pass around in any context, as this can live up to the end of the application if the consumer wants".

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=618322dc1237665746a99dd934347ffb

Reply:
This fully utilizes the trait to make sure it's not getting dropped due to non-use: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=87650c8e735c22878046af4c5d567e69